service/db.go

```diff
@@ -708,3 +708,76 @@ func deleteEmailTokenRow(uuid string) error {
 
 	return nil
 }
+
+// matchEmailAndPassword looks up a row that matches the email. Then after the matched row is retrieved,
+// password retrieved from db is matched with given password.
+// If both email and password matches, returns the matched users row.
+// If the query by email returns nothing, returns email does not exist error.
+// If email is found, but password does not match, returns password does not match error.
+// All other errors are returned.
+func matchEmailAndPassword(email string, password string) (*pblib.User, error) {
+	if err := validateEmail(email); err != nil {
+		return nil, err
+	}
+
+	if err := validatePassword(password); err != nil {
+		return nil, err
+	}
+
+	command := `SELECT uuid, first_name, last_name, email, organization, 
+       				created_timestamp, is_verified, password, permission_level, prospective_email
+				FROM user_svc.accounts 
+				WHERE email = $1
```

Comment:
include `password =  $2`, we are using an indices

Reply:
you will not have to `comparePassword` below

Reply:
we store/retrieve hashed passwords from db, unless the hash will always generate the same string value if i hash user unhashed password??

Reply:
I wonder if we can hash it before sending the query in that way the service is not doing work, and we are utilizing the indices in our db, so we can get rid of this func.

```
// comparePassword compares hashedPassword retrieved from DB and the password from User request.
// Returns nil if match, error if not match or error from bcrypt.
func comparePassword(hashedPassword string, password string) error {
	if hashedPassword == "" || password == "" {
		return consts.ErrInvalidPassword
	}

	err := bcrypt.CompareHashAndPassword([]byte(hashedPassword), []byte(password))
	if err != nil {
		return err
	}

	return nil
}
```

Reply:
package we are using does not do implement that functionality. It is maybe because of security.